gnosis/conditional-markets-interface
# Feature #154: EU KYC Flow

# Description
This adds the first part of the KYC Modalsteps to accommodate **EU Citizens** for our onboarding process.
The following steps are built:
- EU/Non-EU Prompt
- Personal Details
- Pending Screen
- Accepted Screen
- Rejected Screen (Non-SDD Country or application denied)

# To Test
- Follow the flow with a country not permitted for SDD and notice the rejected state
- Follow the flow with a country SDD permitted:
  - Personal Details
    - Notice personal details being validated (E-Mail, Dates)
    - Notice valid data in sight-whitelist database
    - Notice rejection for existing user
    - Notice rejection for duplicate E-Mail
  - Pending Screen
    - Notice automatic updating of pending details
    - Notice automatic redirect to approval or rejection page upon completed process
  - Accepted Screen
    - Notice tier information

# Code
This code adds `react-final-form` as our form handler, as well as a datepicker library. State of the KYC is managed via `useState` - information is gathered from defined `WHITELIST_API`

**⚠️ Notice `WHITELIST_API` URL is now without the `v1` suffix, to allow to decide these values inside the code, as parts of the new application are now required to use `v2`**
